File: code/TS2_RecursiveCall/derived_again/java/DerivedAgainExample.java
package derived_again.java;

/*
+package sandbox
+alias stdout org.eolang.io.stdout
+alias sprintf org.eolang.txt.sprintf

[] > base
  memory > x
  [self v] > n
    x.write v > @
  [self v] > m
    self.n self v > @

[] > derived
  base > @
  [self v] > n
    self.m self v > @

[] > derived_again
  derived > @
*/

class Base {
    public int x;

    public void n(int v) {
        this.x = v;
    }

    public void m(int v) {
        this.n(v);
    }
}

class Derived extends Base {
    @Override
    public void n(int v) {
        this.m(v);
    }
}

class DerivedAgain extends Derived {

}


public class DerivedAgainExample {
    public static void main(String[] args) {
        DerivedAgain derivedAgainInstance = new DerivedAgain();
        derivedAgainInstance.n(12);
    }
}

File: Tests/MutualRecursion/DerivedAgainExample.cpp
class Base {
public:
    int x;

    Base() {
        this->x = 0;
    }

    virtual void n(int v) {
        this->x = v;
    }

    void m(int v) {
        this->n(v);
    }

    virtual ~Base() {}
};

class Derived : public Base {
public:
    virtual void n(int v) override {
        this->m(v);
    }
};

class DerivedAgain : public Derived {

};

int main() {
    Base *derivedAgainInstance = new DerivedAgain();
    derivedAgainInstance->m(12);
    delete derivedAgainInstance;

    return 0;
}
